### PR TITLE
#239/method to add chatroom member to db

### PR DESCRIPTION
**Pull request by Damrod**
as detailed in #239

### CODE REVIEW

File: transcendence-app/src/chat/chat.domain.ts [hunk 1]
```diff
@@ -17,3 +18,15 @@ export class Chat {
     Object.assign(this, entity);
   }
 }
+
+export class ChatMember {
```
Comment (gbudau) [inline transcendence-app/src/chat/chat.domain.ts:22]: praise: I like how you implement this class, looks good :+1:

suggestion: What do we think about moving it to a new file, `chatmember.domain.ts`?

File: transcendence-app/src/chat/dto/create-chatmember.dto.ts [hunk 2]
```diff
@@ -0,0 +1,11 @@
+import { IsNotEmpty, IsString } from 'class-validator';
+
+export class CreateChatMemberDto {
```
Comment (gbudau) [inline transcendence-app/src/chat/dto/create-chatmember.dto.ts:3]: thought: We use `ValidationPipe` and DTOs to [auto-validate](https://docs.nestjs.com/techniques/validation#auto-validation) the incoming request. I see that we are using this class internally, and `ValidationPipe` only validates the controller endpoints.

suggestion: Consider removing this class unless we want to add in the future an endpoint like `POST /chat/room/members` where we specify the `chatId` and `userId` in the request body.

**Review by gbudau — APPROVED** [review]
Thank you for this feature! LGTM :rocket:

File: transcendence-app/src/chat/dto/create-chatmember.dto.ts (hunk 2 above)
Comment (Damrod) [reply transcendence-app/src/chat/dto/create-chatmember.dto.ts:3]: There have already been some [comments](https://github.com/42AGV/ft_transcendence/pull/259#discussion_r1005891878) requesting the addition of this feature. But I do not plan to add it now, and we shouldn't pay for what we don't use. I understand what you are saying. I will remove it.

Comment (Damrod) [reply transcendence-app/src/chat/dto/create-chatmember.dto.ts]: done

File: transcendence-app/src/chat/chat.domain.ts (hunk 1 above)
Comment (Damrod) [reply transcendence-app/src/chat/chat.domain.ts]: done

**Pull request merged**